components/org.wso2.carbon.extension.identity.sso.jwt/src/main/java/org/wso2/carbon/identity/sso/jwt/processor/JWTInboundRequestProcessor.java

```diff
@@ -0,0 +1,669 @@
+                        JWTInboundConstants.ErrorMessages.MISCONFIGURATION_MESSAGE);
+            }
+
+            // Validate JWS Algorithm
+            if (!validateJwsAlgorithm(identityRequest)) {
+                String msg = "Invalid JWT Signing Algorithm configured.";
+                log.error(msg, new JWTIdentityException(msg));
+                return JWTInboundUtil.sendToRetryPage(JWTInboundConstants.ErrorMessages.MISCONFIGURATION_STATUS,
+                        JWTInboundConstants.ErrorMessages.MISCONFIGURATION_MESSAGE);
+            }
+
+            // Set and Validate Redirect URL
+            if (!handleRedirectUrl(identityRequest)) {
+                log.error("Invalid redirect URL: " + neutralize(this.redirectUrl) +
+                        " in the authentication request from the relying party: " + neutralize(this.relyingParty));
+                return JWTInboundUtil.sendToRetryPage(JWTInboundConstants.ErrorMessages.MISCONFIGURATION_STATUS,
+                        JWTInboundConstants.ErrorMessages.MISCONFIGURATION_MESSAGE);
+            }
+
+            // Set and Validate Error URL
+            if (!handleErrorUrl(identityRequest)) {
+                log.error("Invalid error URL: " + neutralize(this.errorUrl) +
+                        " in the authentication request from the relying party: " + neutralize(this.relyingParty));
+                return JWTInboundUtil
+                        .sendToRetryPage(JWTInboundConstants.ErrorMessages.MISCONFIGURATION_STATUS,
+                                JWTInboundConstants.ErrorMessages.MISCONFIGURATION_MESSAGE);
+            }
+
+            if (StringUtils.isNotBlank(logoutRequestPath) &&
+                    identityRequest.getRequestURI().contains(logoutRequestPath)) {
+                // Handle logout request.
+                if (log.isDebugEnabled()) {
+                    log.debug("Handling logout request.");
+                }
+                return buildResponseForFrameworkLogout(messageContext);
+            } else if (StringUtils.isNotBlank(sessionId)) {
+                // A session exists - response coming from the framework after authentication or after logged out.
+                return handleFrameworkResponse(messageContext, identityRequest, respBuilder);
+            } else {
+                // No session exists - send the request to the identity framework.
+                if (log.isDebugEnabled()) {
+                    log.debug("Session not exists. Sending request to identity framework for authentication.");
+                }
+                return buildResponseForFrameworkLogin(messageContext);
+            }
+        } else {
+            String msg = "A Service Provider with the Relying Party '" + neutralize(this.relyingParty) + "' is not " +
+                    "registered. Service Provider should be registered in advance.";
+            log.error(msg, new JWTIdentityException(msg));
```

Comment:
We can make this a debug log as this is actually a client error.

Reply:
A common error will be shown to the client. But if we make it debug, it becomes mandatory to enable debug level logs to understand the actual error on the log.